xwiki-commons-core/xwiki-commons-properties/src/main/java/org/xwiki/properties/internal/DefaultBeanDescriptor.java

```diff
@@ -66,7 +68,9 @@
     /**
      * The properties of the bean.
      */
-    private Map<String, PropertyDescriptor> parameterDescriptorMap = new LinkedHashMap<String, PropertyDescriptor>();
+    private Map<String, PropertyDescriptor> parameterDescriptorMap = new LinkedHashMap<>();
+
+    private Map<PropertyGroup, PropertyGroupDescriptor> groups = new HashMap<>();
```

Comment:
You sure two different instances of PropertyGroup with the same value give the same hashCode/equals ?

Reply:
[This](https://docs.oracle.com/javase/8/docs/api/java/lang/annotation/Annotation.html#equals-java.lang.Object-) + the tests would have failed otherwise.

Reply:
OK it's surprising but that's good to know :)